src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,16 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useAppDispatch } from '../../app/hook';
+import { Status } from '../../types/Status';
+import { actions as filterActions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const addS = (status: Status) => {
+    dispatch(filterActions.addStatus(status));
+  };
+
+  const addQ = (query: string) => {
+    dispatch(filterActions.addQuery(query));
+  };
+
+  const [value, setValue] = useState('');
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    const newValue = e.target.value;
+
+    setValue(newValue);
+    addQ(newValue);
+  }
+
+  function handleClick() {
+    setValue('');
+    addQ('');
+  }
+
+  function handleStatusChange(e: React.ChangeEvent<HTMLSelectElement>) {
+    addS(e.target.value as Status);
```

Comment:
The `handleStatusChange` function casts the value to `Status` using `as Status`. Ensure that the `Status` type includes all possible values from the select options to prevent runtime errors. Consider validating the value before casting.